cbits/tcdrain.c

```diff
@@ -0,0 +1,21 @@
+/* Provide tcdrain() if it is missing (as on some Android versions). */
+
+#include "HsUnixConfig.h"
+#ifndef HAVE_TCDRAIN
+
+#include "ghcplatform.h"
+
+#ifdef linux_android_HOST_OS
+#include <sys/ioctl.h>
+
+int tcdrain(int fd)
```

Comment:
We could rename the function that we import, like we do with `__hsunix_execvpe` (except identifiers containing a double underscore are also reserved, oops!) In principle that would matter if we linked against another library that also noticed that it was missing the same function and provided its own implementation under the same name.

I don't think pushing all the platform-specific logic (including a separate foreign import of `ioctl()`) into Haskell is a cleaner approach. Particularly in the case of `ctermid()` which is imported from two modules which are almost entirely duplicates of each other.